### DISCUSSION TITLE
Icon for undo itmes

### DISCUSSION THREAD

**Discussion by DuckHK** [moudey/Shell#561] [Ideas]
Recently, while customizing icons for various items as outlined in the configuration documentation, I encountered an issue with the "Undo" items (e.g., Undo Paste, Undo Delete, etc.).
Despite my efforts to match their IDs or titles, I’ve been unable to assign icons to these actions.
I kindly request either instructions to resolve this issue or a potential enhancement in a future update to make icon assignments for "Undo" items possible.

**Comment by ghost** [comment]
It is possible that not all of the `undo` ids have been implemented. You can simply try `find` instead:
```js
modify(find='Undo Delete' image=icon.new)
// or maybe something like:
modify(find='撤销删除' image=icon.new) 
```

**Reply by DuckHK** [reply]
Thanks for your help. Actually I tried this in the first place, like I mentioned, titles just don't match anyway, no matter using `find` or `where`.
I have tried `find='Undo Delete' and find='Undo Delete(&U)'`, neither works.

**Comment by RubicBG** [comment]
The `undo` and `redo` commands are an exception - there is one command for each of these two words, but the execution of the command is not always the same and accordingly, the name also changes (not always):

`undo`, `undo copy`, `undo move`, `undo delete`, `undo rename`, `undo new`, `undo apply properties`
`redo`, `redo copy`, `redo move`, `redo delete`, `redo rename`, `redo new`, `redo apply properties`

so use another approach: 

```js
modify(where=str.start(this.title, title.undo) image=[\uE135, #f00] ) // for EN
modify(where=str.contains(this.title, title.undo) image=[\uE135, #ffff00] ) // for ML
modify(find='Undo*' image=icon.undo(auto, color.red) )
modify(find='Undo*|Redo*' image=icon.code ) // change both
```

all examples recognize `&` in the title (tested for 2 languages). It is also possible to remove the `&` before checking, but I have never had to.

**Comment by RubicBG** [comment]
can you make a screenshot of `undo` command?

**Reply by DuckHK** [reply]
`modify(where=str.contains(this.title, title.undo) image=[\uE135, #ffff00] )` this works very well for me and I am able to modify the icons finally!
Thanks a lot!!!!
